src/modality-specific-files/eye-tracking.md

```diff
@@ -0,0 +1,208 @@
+[Common Principles - Definitions](../common-principles.md#definitions) for the
+eye-tracking context.
+
+-   **Eye-tracker** - refers to the apparatus allowing the records of gaze
+    position and/or pupil size.
+
+-   **Screen** - Video display used to present visual stimulation (for example
+    projector, monitor, tablet).
+
+## Eye-tracking data
+
+With EyeTracking-BIDS, we wish to promote the adoption of good practices in
+the management of scientific data.
+
+### Template
+
+<!--
+This block generates a filetree exanple.
+A guide for editing it can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md.
+-->
+{{ MACROS___make_filetree_example(
+
+   {
+   "sub-<label>": {
+      "<datatype>": {
+         "<matches>_eyetrack.json": "",
+         "<matches>_eyetrack.tsv.gz": "",
+         },
+      }
+   }
+
+) }}
+
+Eye-tracking data MUST be stored in the main data recording modality or
+ `<datatype>` directory (for example `func` when combined with fMRI, or
+ `beh` when combined with behavioral measures).
+
+In the template filenames, the `<matches>` part corresponds to task filename
+before the suffix. For example, when fMRI is considered the main modality with
+files such as `sub-01_ses-1_task-pRF_run-1_bold.nii.gz`, `<matches>` would
+then correspond to `sub-01_ses-1_task-pRF_run-1`.
+
+### Tabular file (`*_eyetrack.tsv[.gz]`)
+
+The eye-tracking data must be saved in `.tsv` or `.tsv.gz` format.
+The tabular files consist of one row per sample and a set of REQUIRED and
+OPTIONAL columns.
+
+<!--
+This block generates a columns table.
+The definitions of these fields can be found in
+  src/schema/rules/tabular_data/eyetrack.yaml
+and a guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_columns_table("eyetrack.EyeTracking") }}
+
+Throughout BIDS you can indicate missing values with `n/a` (for "not
+available").
+
+We encourage users to put the eye-tracking raw data within the
+`/sourcedata` directory (for example put eye-tracker proprietary files before
+conversion to `tsv.gz` format in the sourcedata directory).
+
+### Sidecar JSON document (`*_eyetrack.json`)
+
+#### Required fields
+
+The following specific eye-tracking fields MUST be present:
+
+<!-- This block generates a metadata table.
+These tables are defined in
+ src/schema/rules/sidecars
+The definitions of the fields specified in these tables may be found in
+ src/schema/objects/metadata.yaml
+A guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_sidecar_table("eyetrack.EyeTrackingRequired") }}
+
+Note that the following fields from `*_events.json` files
+are REQUIRED as they are considered essential in eye-tracking data analysis:
+-   `StimulusPresentation.ScreenSize`,
+-   `StimulusPresentation.ScreenResolution`,
+-   `StimulusPresentation.ScreenDistance`.
```

Comment:
Not sure how to encode this in the schema.

The rule would be something like.

if there is a file [match]_eyetrack.json 
in the same folder as the data file [match]_{suffix_for_this_dataype}.ext, 
then in the sidecar file [match]_events.json,
the following subfields are required:
-   `StimulusPresentation.ScreenSize`
-   `StimulusPresentation.ScreenResolution`
-   `StimulusPresentation.ScreenDistance`

changes must be added in `src/schema/rules/sidecars/events.yml` but I am not sure how to express this in the schema

I think this is the first case of making a subfield required, no?

@effigies @tsalo @rwblair @nellh 

This is not urgent but will have to be adressed at one point.